sdcm/utils/benchmarks.py

```diff
@@ -0,0 +1,137 @@
+        self._install_ubuntu_prerequisites()
+        self._build_and_install_sysbench()
+        return self._check_sysbench_installation()
+
+    def _install_ubuntu_prerequisites(self) -> bool:
+        package_list = ["make", "automake", "libtool", "pkg-config", "libaio-dev"]
+        try:
+            LOGGER.info("Installing Ubuntu prerequisites for the node benchmarks...")
+            for pkg in package_list:
+                self._node.install_package(pkg)
+            return True
+        except Exception:
+            LOGGER.warning("Failed to install Ubuntu prerequisites for the node benchmarking tools.")
+            raise
+
+    def _build_and_install_sysbench(self) -> bool:
+        build_and_install_script = """\
+            cd ./sysbench
+            ./autogen.sh
+            ./configure --without-mysql
+            make -j
+            make install
+        """
+        try:
+            result = self._remoter.sudo(shell_script_cmd(build_and_install_script))
+            LOGGER.debug("Build and install sysbench result: %s", result.stdout)
+            LOGGER.debug("Build and install sysbench errors: %s", result.stderr)
+            return True
+        except Exception:
+            LOGGER.warning("Failed to install sysbench from source.")
+            raise
+
+    def _clone_sysbench_repo(self):
+        try:
+            LOGGER.info("Cloning sysbench repo...")
+            clone_cmd = "git clone https://github.com/akopytov/sysbench.git"
+            self._remoter.sudo(clone_cmd)
+            LOGGER.info("Finished cloning sysbench repo.")
+        except Exception:
+            LOGGER.warning("Failed to clone sysbench repo.")
+            raise
+
+    def _check_sysbench_installation(self) -> bool:
+        check_cmd = "sysbench --version"
+        LOGGER.info("Checking sysbench installation...")
+        result = self._remoter.run(check_cmd)
+        if result.stdout and not result.stderr:
+            LOGGER.info("Check: sysbench installation version: %s", result.stdout)
+            return True
+        return False
+
+    def run_sysbench(self, thread_num: int = None, test_time: int = 120) -> str:
+        thread_num = thread_num if thread_num else self._get_number_of_cores_on_node()
+        run_cmd = f"sysbench cpu --threads={thread_num} --time={test_time} run"
+        sysbench_run_result = self._remoter.run(run_cmd)
+        if sysbench_run_result.stderr:
+            LOGGER.warning("sysbench benchmark ran with errors: \n%s", sysbench_run_result.stderr)
+        LOGGER.info("Sysbench results:")
+        LOGGER.info("%s", sysbench_run_result.stdout)
+        return sysbench_run_result.stdout
+
+    def _get_number_of_cores_on_node(self) -> int:
+        cat_cmd = "cat /proc/cpuinfo"
+        regex_pattern = re.compile(r"cpu cores\s+:\s+(?P<core_num>\d+)")
+        cat_result = self._remoter.run(cat_cmd).stdout
+        core_num = int(regex_pattern.search(cat_result).groupdict()["core_num"])
```

Comment:
It doesn't look safe. If for some reason search don't find anything then we get python errors such as 'KeyError' and 'TypeError'.

Reply:
I agree that it's not safe, but I'd rather have it fail in a case where it can't get the data.

Reply:
We have property on basenode 
   @property
    def cpu_cores
Why not use it?

Reply:
Why not use cpu_cores property?